book/src/developers/designs/2025-07-entra-id.md

```diff
@@ -0,0 +1,161 @@
+# Entra ID / Azure AD compatability proposal
+
+A proposal has been made by [SerNET](https://www.sernet.com/) that they wish to invest in the building of an Entra ID /
+Azure AD compatible IDM service. This may be initially supported be EU grant funding.
+
+David Mulder, the creator of the [himmelblau](https://github.com/himmelblau-idm/) project has proposed that Kanidm could
+be the IDM service backend, where an Entra ID compatible authentication module would be added.
+
+Himmelblau and Kanidm already have many connections and projects and collaborators. As himmelblau is already a working
+client to Entra ID, and Kanidm is a full fledged IDM, this existing collaboration and relationship makes it a natural
+fit that we could work together on this proposal.
+
+In this document, we want to examine the pros, cons, and risks of this proposal.
+
+## Goal of the Proposal
+
+The goal would be to add compatibility for Kanidm to allow Himmelblau as an authentication client. If Himmelblau is able
+to function as a client to Kanidm, this demonstrates that an opensource Entra ID replacement is viable.
+
+With a viable opensource replacement available, this may allow regulatory pressure to be exerted on Microsoft to allow
+open Windows to use third party Entra ID compatible services.
+
+This is important as today Windows is coded to only support Microsofts EntraID services and uses TLS CA pinning to
+ensure that a 3rd party can't MITM this process. This limits Windows authentication options for users to EntraID or on
+prem ADDC.
+
+## Pros
+
+### Improved collaboration between Himmeblau and Kanidm
+
+The Himmelblau project uses a lot of Kanidm source code. By having an Entra ID compatible server, we would be able to
+merge and collabore more to "shrink the gap" between our projects. This would also give Himmelblau improved access to
+client testing since they would be able to test against a provider.
+
+### Kanidm gains client machine support
+
+Kanidm has wanted domain-joined machine support for a long time - this would allow Kanidm to use the existing
+capabilities of Himmeblau rather than inventing our own specification for this. Himmelblau also has some really cool
+native desktop integrations too!
```

Comment:
It'd be good to expand on what the desktop integrations are or provide for usershere

Reply:
The DBUS session service has other uses also. I've got users who are implementing custom SSO bits using the tokens they get from it.